include/deal.II/grid/grid_tools.h

```diff
@@ -2651,6 +2654,39 @@ namespace GridTools
     const std::vector<BoundingBox<spacedim>> &local_bboxes,
     MPI_Comm                                  mpi_communicator);
 
+  namespace bg  = boost::geometry;
+  namespace bgi = boost::geometry::index;
+
+  template <int spacedim>
+  using bgPoint = bg::model::point<double, spacedim, bg::cs::cartesian>;
+
+  template <int spacedim>
+  using bgBox = bg::model::box<bgPoint<spacedim>>;
+
+  template <int spacedim, int leaf_number>
+  using LinearPointRtree = bgi::rtree<std::tuple<bgBox<spacedim>, unsigned int>,
+                                      bgi::linear<leaf_number>>;
+
```

Comment:
This should really be `using LinearBoxRtree = bgi::rtree<BoundingBox<spacedim>, bgi::linear<leaf_number>>`. In this way the rtree algorithm can use anything that has a `BoundingBox<spacedim>` as first element of the tuple/pair, or just a box. 

Notice the name. You copied this from the test where I was creating a tree of points (hence that name: `LinearPointRTree`), but you want t tree of boxes...

Reply:
From boost documentation:


> The default IndexableGetter can translate all types adapted to Point, Box or Segment concepts (called Indexables). Furthermore, it can handle std::pair<Indexable, T>, boost::tuple<Indexable, ...> and std::tuple<Indexable, ...> when possible. For example, for Value of type std::pair<Box, int>, the default IndexableGetter translates from std::pair<Box, int> const& to Box const&.

Reply:
if you define the rtree with indexable `BoundingBox`, your code further down will still work.